### PR TITLE
more unique functions

### PR DESCRIPTION
**Pull request by matthiasdiener**
_Please squash_

### CODE REVIEW

File: pytools/__init__.py [hunk 1]
```diff
@@ -2997,6 +3003,49 @@ def unique(seq: Iterable[T]) -> Iterator[T]:
     :func:`itertools.groupby` (which removes consecutive duplicates)."""
     return iter(dict.fromkeys(seq))
 
+
+def unique_difference(*args: Iterable[T]) -> Iterator[T]:
```
Comment (inducer) [inline pytools/__init__.py:3007]: Would it make sense to have these return `Collection[T]` (and skip the `iter` call)?

Comment (matthiasdiener) [reply pytools/__init__.py:3007]: Hmm, I'm not sure what to return instead of the `iter` - a list or tuple? We can't just return the dictionary I guess.

Comment (matthiasdiener) [reply pytools/__init__.py]: 9ece452 is a version of these functions returning lists. I don't know if that's better.

Comment (inducer) [reply pytools/__init__.py]: No, I thought you would just return the dictionary. A dictionary with key type `T` should count as a collection of type `T` I think.

Comment (inducer) [reply pytools/__init__.py]: I.e. no need to build a separate list.

Comment (matthiasdiener) [reply pytools/__init__.py]: What do you think of [7b352f3](https://github.com/inducer/pytools/pull/257/commits/7b352f3103f55b627cbf550d404b408638f05354)?

Comment (inducer) [reply pytools/__init__.py]: LGTM!

**Comment by inducer** [comment]
Thx!

**Pull request merged**